plugins/arCasPlugin/lib/casUser.class.php

```diff
@@ -122,7 +132,7 @@ private function setGroupMembership($user, $attributeToCheck, $expectedValue, $g
     {
       // CAS attributes say the user does not belong to the group, so remove
       // them if they are currently a member.
-      if ($user->hasGroup($groupID))
+      if (in_array($groupID, $currentAclUserGroupIds))
```

Comment:
In the edge case where:
- a user assigned to a group is removed from a group on the CAS side, 
- and the CAS group is valid inside app_cas_user_groups setting 
- and the group is still linked to the user in AtoM, 
...will the group be successfully removed during CAS authentication? 

A second edge case:
- a user is currently assigned to a group
- they authenticate via CAS which also has them in that group
- the group has been removed from the 'app_cas_user_groups' setting
Will the invalid group be successfully removed from the user?

Reply:
> In the edge case where:
> 
>     * a user assigned to a group is removed from a group on the CAS side,
> 
>     * and the CAS group is valid inside app_cas_user_groups setting
> 
>     * and the group is still linked to the user in AtoM,
>       ...will the group be successfully removed during CAS authentication?

Yes, exactly. If a user is downgraded on the CAS server and no configuration change has been made in AtoM, AtoM will reflect the change and update the group membership on the next login.

> A second edge case:
> 
>     * a user is currently assigned to a group
> 
>     * they authenticate via CAS which also has them in that group
> 
>     * the group has been removed from the 'app_cas_user_groups' setting
>       Will the invalid group be successfully removed from the user?

Interesting.This edge case is not currently handled by the code. As-is, it would currently require an administrator to manually remove the user group that's not longer configured in `app_cas_user_groups` from users who belong to it. But I think we could accomplish this by implementing logic that's a bit closer to the `user` module's `editAction`, whereby a user's group membership is reset on each login and only appropriately configured groups are included.